utils/db_utils.py

```diff
@@ -15,6 +15,16 @@
 from utils import permissions as perm_utils
 from utils.datetime_utils import iso_range_to_ts_range
 
+def df_to_filtered_records(df, col_to_filter=None, vals_to_exclude: list[str] = []):
+    """
+    Returns a dictionary of df records, given a dataframe, a column to filter on,
+    and a list of values that rows in that column will be excluded if they match
+    """
+    if df.empty: return []
+    if col_to_filter and vals_to_exclude:
```

Comment:
this should avoid the filtering operation when there are no vals to exclude.

```suggestion
    if col_to_filter and vals_to_exclude and len(vals_to_exclude) > 0:
```

Reply:
I read that empty lists are falsy in Python, so I thought we wouldn't have to specifically check the length.

Do you think it's better to just be explicit?

Reply:
I did not know that, but I checked and your reading was correct.

```
>>> t = []
>>> if t:
...     print("true")
...
>>> if not t:
...     print("false")
...
false
```

Interesting! That was non-obvious to me (at least), so I would suggest making it explicit or adding a comment.

Reply:
added comment